security/security.c

```diff
@@ -307,6 +382,49 @@ static void __init ordered_lsm_parse(const char *order, const char *origin)
 	kfree(sep);
 }
 
+static void __init lsm_init_hook_static_slot(struct security_static_slot *slots,
+					     struct hlist_head *head,
+					     int *first_slot_idx)
+{
+	struct security_hook_list *pos;
+	struct security_static_slot *slot;
+	int slot_cnt, first_slot;
+
+	slot_cnt = 0;
+	hlist_for_each_entry_rcu(pos, head, list) {
+		slot_cnt++;
+	}
+
+	first_slot = SECURITY_STATIC_SLOT_COUNT - slot_cnt;
+	if (first_slot < 0)
+		panic("%s - No static hook slot remaining to add LSM hook.\n",
+		      __func__);
+
+	if (slot_cnt == 0) {
+		*first_slot_idx = INT_MAX;
+		return;
+	}
+
+	slot = slots + first_slot;
+	hlist_for_each_entry_rcu(pos, head, list) {
+		__static_call_update(slot->key, slot->tramp,
+				     pos->hook.generic_func);
+		slot++;
+	}
+
+	*first_slot_idx = first_slot;
```

Comment:
I find the "first_slot" name confusing because it feels like it refers to a slot rather than an index.
I wonder if you could maybe simplify this code by writing to *first_slot_idx straight away, and then only resetting *first_slot_idx to INT_MAX in the failing case. this way, the variable name would be clearer and you save a line of code.

Reply:
Good point. I removed `first_slot` to just use the pointer instead.